test/dynamo/test_unspec.py

```diff
@@ -601,6 +601,20 @@ def fn(x):
         compl_fn = torch.compile(fn, dynamic=True, backend="eager")
         self.assertEqual(compl_fn(inputs), fn(inputs))
 
+    @torch._dynamo.config.patch(specialize_float=False)
+    def test_unspec_roundtrip_float_input(self):
+        def f(x, y):
+            if y == 5.0:
+                return x + 2
+            else:
+                return x + y
+            return (x, y)
+
+        cf = torch.compile(backend="eager", fullgraph=True)(f)
+        x = 1.1234567891234568
+        y = 1.1234567891234569
+        self.assertAlmostEqual(f(x, y), cf(x, y))
```

Comment:
I was shocked to find that if I did `assertEqual` here instead of `assertAlmostEqual`, the test would erroneously pass without my fix. I still haven't found a good explanation as to why, since it seems like under the hood `assertEqual` should use `==`.

If you print out `print(f(x, y), cf(x, y))` you will see they differ.

If you assertTrue(f(x, y) == cf(x, y)) it will fail

But if you `assertEqual(f(x, y), cf(x, y))` it will pass...

Reviewers, if you happen to know and could enlighten me on what's going on, that'd be much appreciated!

Reply:
assertEqual has built in rtol/atol. If you want to test bitwise equivalence use frexp